xbmc/cores/VideoPlayer/VideoRenderers/DebugRenderer.cpp

```diff
@@ -6,178 +6,150 @@
+{
+  if (!m_isInitialized)
+    return;
+
+  // FIXME: We currently force ASS_Event's and the current PTS
+  // of rendering with fixed values to allow perpetual
+  // display of on-screen text. It would be appropriate for Libass
+  // provide a way to allow fixed on-screen text display
+  // without use all these fixed values.
+  m_adapter->FlushSubtitles();
+  m_adapter->AddSubtitle(video.videoSource.c_str(), 0., 5000000.);
+  m_adapter->AddSubtitle(video.metaPrim.c_str(), 0., 5000000.);
+  m_adapter->AddSubtitle(video.metaLight.c_str(), 0., 5000000.);
+  m_adapter->AddSubtitle(video.shader.c_str(), 0., 5000000.);
+  m_adapter->AddSubtitle(render.renderFlags.c_str(), 0., 5000000.);
+  m_adapter->AddSubtitle(render.videoOutput.c_str(), 0., 5000000.);
 }
 
 void CDebugRenderer::Render(CRect& src, CRect& dst, CRect& view)
 {
+  if (!m_isInitialized)
+    return;
+
   m_overlayRenderer.SetVideoRect(src, dst, view);
   m_overlayRenderer.Render(0);
 }
 
 void CDebugRenderer::Flush()
 {
-  m_overlayRenderer.Flush();
+  if (!m_isInitialized)
+    return;
+
+  m_adapter->FlushSubtitles();
 }
 
 CDebugRenderer::CRenderer::CRenderer() : OVERLAY::CRenderer()
 {
-  m_font = "__debug__";
-  m_fontBorder = "__debugborder__";
 }
 
 void CDebugRenderer::CRenderer::Render(int idx)
 {
-  std::vector<COverlay*> render;
   std::vector<SElement>& list = m_buffers[idx];
-  float posY = 0.0f;
   for (std::vector<SElement>::iterator it = list.begin(); it != list.end(); ++it)
   {
     COverlay* o = nullptr;
 
     if (it->overlay_dvd)
-      o = Convert(it->overlay_dvd, it->pts);
-
-    if (!o)
-      continue;
+    {
+      CDVDOverlayLibass* ovAss = static_cast<CDVDOverlayLibass*>(it->overlay_dvd);
+      if (!ovAss || !ovAss->GetLibassHandler())
+        continue;
 
-    COverlayText *text = dynamic_cast<COverlayText*>(o);
-    if (text)
-      text->PrepareRender("arial.ttf", 1, 100, 15, 0, m_font, m_fontBorder, UTILS::COLOR::NONE,
-                          m_rv);
+      KODI::SUBTITLES::style subStyle;
+      bool updateStyle = !m_debugOverlayStyle;
+      if (updateStyle)
+        CreateSubtitlesStyle();
 
-    RESOLUTION_INFO res = CServiceBroker::GetWinSystem()->GetGfxContext().GetResInfo(CServiceBroker::GetWinSystem()->GetGfxContext().GetVideoResolution());
+      o = ConvertLibass(ovAss, it->pts, updateStyle, m_debugOverlayStyle);
 
-    o->m_pos = COverlay::POSITION_ABSOLUTE;
-    o->m_align = COverlay::ALIGN_SCREEN;
-    o->m_x = 10 + (o->m_width * m_rv.Width() / res.iWidth) / 2;
-    o->m_y = posY + o->m_height;
-    OVERLAY::CRenderer::Render(o, 0);
+      if (!o)
+        continue;
+    }
 
-    posY = o->m_y;
+    OVERLAY::CRenderer::Render(o);
   }
-
   ReleaseUnused();
 }
+
+void CDebugRenderer::CRenderer::CreateSubtitlesStyle()
+{
+  m_debugOverlayStyle = std::make_shared<KODI::SUBTITLES::style>();
+  m_debugOverlayStyle->fontName = "arial.ttf";
+  m_debugOverlayStyle->fontSize = 16.0;
```

Comment:
```suggestion
  m_debugOverlayStyle->fontSize = 20.0;
```
After tested with 4K screen I agree with you that 16 is too small. At this time I think 20 is a good size 😃

Reply:
hahaha ok will do